lemur/plugins/lemur_gcp/plugin.py

```diff
@@ -48,22 +45,23 @@ class GCPDestinationPlugin(DestinationPlugin):
     ]
 
     def upload(self, name, body, private_key, cert_chain, options, **kwargs):
-
         try:
             ssl_certificate_body = {
-                "name": self._certificate_name(body),
+                "name": certificates.get_name(body),
                 "certificate": body,
                 "description": "",
                 "private_key": private_key,
             }
-            credentials = self._get_gcp_credentials(options)
+            credentials = auth.get_gcp_credentials(self, options)
             return self._insert_gcp_certificate(
                 self.get_option("projectID", options),
                 ssl_certificate_body,
                 credentials,
             )
 
         except Exception as e:
+            if "already exists" in str(e):
```

Comment:
The python SDK should be raising an `AlreadyExists` exception (see [docs](https://googleapis.dev/python/google-api-core/latest/exceptions.html)) which can be leveraged to handle this in a more pythonic way.

```python3
from google.api_core import exceptions

try:
    self._insert_gcp_certificate(
        self.get_option("projectID", options),
        ssl_certificate_body,
        credentials,
    )
except exceptions.AlreadyExists as e:
    # certificate already exists
```